lib/rules/sotd/charter-disclosure.js

```diff
@@ -1,15 +1,81 @@
+
+        // check text exists
         var txt = sr.norm($sotd.text());
-        if (!/The disclosure obligations of the Participants of this group are described in the charter\./.test(txt))
-            sr.error(self, "not-found");
+        if (!charterText.test(txt)) {
+            sr.error(self, "text-not-found");
+            return done();
+        }
+
+        // check "charter" link is found and correct
+        var charterLinkFound = false;
+        $sotd.find("a[href]").each(function () {
+            var $a = sr.$(this)
+                , charterHref = $a.attr("href")
+                , text = sr.norm($a.text())
+                , pText = sr.norm($a.parent().text())
+                ;
+
+            // Find the right paragraph and right link.
+            if (charterText.test(pText) && text == "charter") {
+                charterLinkFound = true;
+                
+                var expectedHref
+                    , deliverId = sr.getDelivererIDs()
+                    , url = 'https://api.w3.org/groups/' + deliverId[0] + '/charters'
+                    , req
+                    , apikey = process.env.W3C_API_KEY
+                    , ua = "W3C-Pubrules/" + sr.version
+                    ;
+                if (deliverId.length === 0) {
+                    sr.error(self, "no-group");
+                    return done();
+                }
+                req = sua.get(url)
+                    .set("User-Agent", ua);
+                req.query({ apikey: apikey, embed: true});
+                req.end(function (err, res) {
+                    if (err || !res.ok) {
+                        sr.error(self, "no-response", { status: (res ? res.status : (err ? err : 'error')) });
+                    } else {
+                        // request for charter url (to request for expected charter)
+                        var charters = res.body && res.body._embedded.charters
+                            , docDate = sr.getDocumentDate();
+
+                        // there can be more than on charters, find the charter that match the document publication date
+                        charters.forEach(element => {
+                            var endDate = new Date(element.end);
+                            var startDate = new Date(element.start);
+
+                            if (docDate < endDate && docDate > startDate) {
+                                expectedHref = element.uri + '#patentpolicy';
+                            }
+                        });
+
+                        if (expectedHref !== charterHref) {
+                            sr.error(self, "wrong-link", { "expectedHref": expectedHref });
+                        }
+                        done();
+                    }
+                });
+                return false; // break $.each after find charter in document
+            }
+        });
+        if (!charterLinkFound) {
+            sr.error(self, "link-not-found");
+            done();
+        }
```

Comment:
```suggestion
        $sotd.find("a[href]").each(function () {
            var $a = sr.$(this)
                , charterHref = $a.attr("href")
                , text = sr.norm($a.text())
                , pText = sr.norm($a.parent().text())
                ;

            // Find the right paragraph and right link.
            if (charterText.test(pText) && text === "charter") {
                charterLinkFound = true;
                
                var expectedHref
                    , deliverId = sr.getDelivererIDs()
                    , url = 'https://api.w3.org/groups/' + deliverId[0] + '/charters'
                    , req
                    , apikey = process.env.W3C_API_KEY
                    , ua = "W3C-Pubrules/" + sr.version
                    ;
                if (deliverId.length === 0) {
                    sr.error(self, "no-group");
                    return false;
                }
                req = sua.get(url)
                    .set("User-Agent", ua);
                req.query({ apikey: apikey, embed: true});
                req.end(function (err, res) {
                    if (err || !res.ok) {
                        sr.error(self, "no-response", { status: (res ? res.status : (err ? err : 'error')) });
                    } else {
                        // request for charter url (to request for expected charter)
                        var charters = res.body && res.body._embedded.charters
                            , docDate = sr.getDocumentDate();

                        // there can be more than on charters, find the charter that match the document publication date
                        charters.forEach(element => {
                            var endDate = new Date(element.end);
                            var startDate = new Date(element.start);

                            if (docDate < endDate && docDate > startDate) {
                                expectedHref = element.uri + '#patentpolicy';
                            }
                        });

                        if (expectedHref !== charterHref) {
                            sr.error(self, "wrong-link", { "expectedHref": expectedHref });
                        }
                    }
                });
                return false; // break $.each after find charter in document
            }
        });
        if (!charterLinkFound) {
            sr.error(self, "link-not-found");
        }
        done();
```

Reply:
`req.end` in `$sotd.find("a[href]").each()`  is called asynchronously, so if moving `done()` to the last line , `done()` will be called earlier than function in  `req.end`